src/ui/SimpleFilter/SimpleFilter.ts

```diff
@@ -458,6 +471,23 @@ export class SimpleFilter extends Component {
     return this.circleElement.el;
   }
 
+  public buildClearElement(): HTMLElement {
+    this.clearElement = $$('button', { className: 'coveo-simplefilter-eraser' }, SVGIcons.icons.mainClear);
+    this.clearElement.hide();
+
+    new AccessibleButton()
+      .withElement(this.clearElement.el)
+      .withLabel(l('Clear', this.options.title))
+      .withClickAction((evt: Event) => {
+        evt.stopPropagation();
+        this.handleClear();
+      })
+      .withEnterKeyboardAction(() => this.handleClear())
```

Comment:
Since you are using an actual `button` element, these are superfluous. The `AccessibleButton` class is mostly meant to transform "non button" element (like, a div or a span) to act like a normal button should (click == enter on keyboard, label, etc).

We have this class in place for backward compatibility reason, where we can't go and change an existing element (that was a div) to a button, so we bridge the gap like this.

In this case, since it's a new option/new markup, I would not use this class at all, and just create a normal button with all correct properties (the label), and it will be accessible by default.

Reply:
Cool